enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,336 @@
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+    - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+    - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+#### Immutable API
+
+While the API is immutable, updating to a new version (upgrade or
+downgrade) is safe because nothing will change the values in the
+`status` block of the resource.
+
+#### Mutable API (future work)
+
+Upgrading the API to add new fields is safe because old consumers will
+ignore them until they are also upgraded.
+
+Adding new values to enum fields is not safe, because there is a
+window of time between when the `capabilities` API is updated and when
+the operators using the API are updated. To avoid issues during that
+time, the controller that manages the capabilities API will wait for
+all `ClusterOperator` resources to have the same version and not
+report their status as progressing, before it copies settings from the
+`spec` fields to the `status` used by consumers. This will ensure that
+the other operators will be upgraded and start consuming the old
+settings. Then, when all of the other operators are updated, the
+settings in the `status` fields will be updated and the other
+operators will re-configure their operands.
+
+For a downgrade, the administrator may need to manually update the
+`capabilities` API to change settings that use new values not
+supported in the older version.
+
+### Version Skew Strategy
```

Comment:
@mrunalp @staebler I added some details about handling version skew based on our conversation earlier today. Let me know if I've captured it accurately here.

Reply:
@dhellmann Looks good. Thanks!